src/main/java/com/alistats/discorki/listener/SlashCommandListener.java

```diff
@@ -0,0 +1,200 @@
+                    break;
+                case "list_users":
+                    listUsers(event);
+                    break;
+                case "list_notifications":
+                    event.reply("This feature is still in development.").queue();
+                    break;
+                case "leaderboard":
+                    leaderboard(event);
+                    break;
+            }
+        } catch (Exception e) {
+            event.reply("An error occured.").queue();
+            e.printStackTrace();
+        }
+
+    }
+
+    private void add(SlashCommandInteractionEvent event) throws Exception {
+        // Get guild
+        DiscordGuild discordGuild = getOrCreateGuild(event.getGuild());
+        DiscordUser discordUser = new DiscordUser();
+
+        User user = event.getOption("user").getAsUser();
+        String summonerName = event.getOption("summoner_name").getAsString();
+
+        // Check if user is already in that guild
+        if (discordGuild.getUsers() != null
+                && discordGuild.getUsers().stream().anyMatch(u -> u.getId().equals(user.getId()))) {
+            // Get user
+            discordUser = discordGuild.getUsers().stream().filter(u -> u.getId().equals(user.getId())).findFirst()
+                    .get();
+        } else {
+            // Create user
+            discordUser.setId(user.getId());
+            discordUser.setUsername(user.getName());
+            discordUser.setDiscriminator(user.getDiscriminator());
+            discordUser.setGuild(discordGuild);
+            discordUser = userRepo.save(discordUser);
+        }
+
+        // Check if user already has that summoner
+        if (discordUser.getSummoners() != null
+                && discordUser.getSummoners().stream().anyMatch(s -> s.getName().equals(summonerName))) {
+            event.reply("That summoner is already added to that user.").queue();
+        } else {
+            // Fetch summoner details
+            SummonerDto summonerDto = leagueApiController.getSummoner(summonerName);
+            Summoner summoner = summonerDto.toSummoner();
+            summoner.setTracked(true);
+            summonerRepo.save(summoner);
+
+            // Fetch rank
+            List<LeagueEntryDto> leagueEntryDtos = Arrays
+                    .asList(leagueApiController.getLeagueEntries(summoner.getId()));
+
+            // Save entries
+            for (LeagueEntryDto leagueEntryDto : leagueEntryDtos) {
+                rankRepo.save(leagueEntryDto.toRank(summoner));
+            }
+
+            // Add summoner to user
+            discordUser.addSummoner(summoner);
+            userRepo.save(discordUser);
+            event.reply("Added summoner to user.").queue();
+        }
+    }
+
+    private void remove(SlashCommandInteractionEvent event) {
+        try {
+            userRepo.deleteById(event.getOption("user").getAsUser().getId());
+            event.reply("Removed user from guild.").queue();
```

Comment:
I don't know what a guild is, but does it remove from a guild or should the message say:
`Stopped tracking user`?

Reply:
```suggestion
            event.reply("Removed user from being tracked.").queue();
```